src/combinatorics/simple_graph/clique.lean

```diff
@@ -142,6 +142,27 @@ forall_imp $ λ s, mt $ is_n_clique.mono h
 
 end clique_free
 
+/-! ### Set of cliques -/
+
+section clique_set
+variables (G) {n : ℕ} {a b c : α} {s : finset α}
+
+/-- The `n`-cliques in a graph as a set. -/
+def clique_set (n : ℕ) : set (finset α) := {s | G.is_n_clique n s}
+
+lemma mem_clique_set_iff : s ∈ G.clique_set n ↔ G.is_n_clique n s := iff.rfl
+
+@[simp] lemma clique_set_eq_empty_iff : G.clique_set n = ∅ ↔ G.clique_free n :=
+by simp_rw [clique_free, set.eq_empty_iff_forall_not_mem, mem_clique_set_iff]
+
+alias clique_set_eq_empty_iff ↔ _ simple_graph.clique_free.clique_set
+
+attribute [protected] clique_free.clique_set
+
+lemma clique_set_mono (h : G ≤ H) : G.clique_set n ⊆ H.clique_set n := λ _, is_n_clique.mono h
```

Comment:
Do you want `G.clique_set ≤ H.clique_set` as well? Also `mono` tag

Reply:
I don't think we care about `G.clique_set ≤ H.clique_set`. It's a weird spelling of the above.

Reply:
What more general version? `G.clique_set ≤ H.clique_set ↔ ∀ n, G.clique_set n ⊆ H.clique_set n`, right?

Reply:
I don't believe that's the question - I'm explicitly not saying you should remove this version, and that you should instead add the more generally applied `G.clique_set ≤ H.clique_set`. Do you think it is harmful to have the extra lemma?

Reply:
I think it sets a bad precedent, because I've quite literally never seen such a lemma be useful anywhere. I don't know what you mean by "more generally applied".

Reply:
> I've quite literally never seen such a lemma be useful anywhere

This comes down heavily to personal experience, I've used versions like `p ≤ q` orders of magnitude more than the applied versions; in particular when working with order theoretic definitions on functions, and if only the `∀ a, p a ≤ q a` version is present one needs to construct lambda terms inline in order to use them, which as I mentioned before, breaks rewrite chains. If anything, it's a bad precedent not to have both versions!